src/features/filter.ts

```diff
@@ -1,10 +1,46 @@
-export const actions = { /* put action creators here */};
+import { Filter } from '../types/Filter';
+import { Sort } from '../types/Sort';
 
-const filterReducer = () => {
-  return {
-    query: '',
-    status: 'all',
-  };
+type QueryAddAction = { type: 'filter/QUERYADD', payload: string };
+type QueryClearAction = { type: 'filter/QUERYCLEAR' };
+type SortChangeAction = { type: 'filter/SORTCHANGE', payload: string };
+
+type Action = QueryAddAction | QueryClearAction | SortChangeAction;
+
+const queryAdd = (value: string): QueryAddAction => ({
+  type: 'filter/QUERYADD',
+  payload: value,
+});
+
+const queryClear = (): QueryClearAction => ({
+  type: 'filter/QUERYCLEAR',
+});
+
+const sortChange = (value: string) => ({
+  type: 'filter/SORTCHANGE',
+  payload: value,
+});
+
+export const actions = { queryAdd, queryClear, sortChange };
+
+const filterReducer = (filter: Filter = {
+  query: '',
+  sort: Sort.all,
+},
+action: Action) => {
+  switch (action.type) {
+    case 'filter/QUERYADD':
+      return { ...filter, query: action.payload };
+
+    case 'filter/QUERYCLEAR':
+      return { ...filter, query: '' };
+
+    case 'filter/SORTCHANGE':
+      return { ...filter, sort: action.payload as Sort };
```

Comment:
Avoid using 'as' for type casting. It's better to handle the type checking in the action creators to ensure type safety. Consider changing the payload type of 'SortChangeAction' to 'Sort' to avoid this.